app/src/test/java/io/gnosis/safe/ui/safe/terms/TermsCheckerTest.kt

```diff
@@ -0,0 +1,55 @@
+package io.gnosis.safe.ui.safe.terms
+
+import android.app.Application
+import io.gnosis.safe.ui.safe.terms.TermsChecker.Companion.TERMS_AGREED
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.spyk
+import junit.framework.Assert.assertFalse
+import junit.framework.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import pm.gnosis.svalinn.common.PreferencesManager
+import pm.gnosis.tests.utils.TestPreferences
+
+class TermsCheckerTest {
+    private lateinit var preferences: TestPreferences
+    private lateinit var preferencesManager: PreferencesManager
+
+    @Before
+    fun setup() {
+        preferences = spyk()
+        val application = mockk<Application>().apply {
+            every { getSharedPreferences(any(), any()) } returns preferences
```

Comment:
There is a test util for shared preferences in svalinn that would be better suited for this. I would also ask you to verify mock calls as they are used. Check the mockk documentation and tests that are already merged so you get a clearer idea of what would be nice to have

Reply:
I tried to use the same way as `SafeRepositoryTest`.